Sources/tinys3/model/Multipart/PartSizeCalculator.swift

```diff
@@ -2,6 +2,6 @@ import Foundation
 
 struct PartSizeCalculator {
     static func calculate(basedOn fileSize: Int) -> Int {
-        max(min(fileSize, 5_000_000), min(4_900_000_000, fileSize / 12))
+        max(min(fileSize, 5_000_000), min(4_900_000_000, fileSize / 50))
```

Comment:
Updated to split most uploads in 50 parts instead of 12, to make the resuming more useful.

Our typical VMs `.aar` files are usually around 40GB, making `fileSize/50 ≈ 800MB`, compared to `fileSize/12 ≈ 3.4GB`. So that means if an upload fails / get interrupted, we'll have more chances that many 800MB parts would have done uploading by then and could be skipped on resume / second try, and the ones to retry will be smaller amount to re-upload.

Reply:
Given the fact that we read this all into memory in `uploadPart`, I wonder if we should err on the side of making this even smaller? 800MB is a lot of data to read into RAM but WDYT?

Reply:
True, I didn't consider the RAM footprint (and now that I think about it, it's crazy to think that we already loaded whole 3.4GB chunks into RAM before that change!).
I'm happy to experiment with smaller parts.

---

Another option to reduce memory footprint would be to check if we really need to whole load the data into RAM in the first place in all the various places we manipulate that part's data. Maybe either using `mmap` (if it's not already used automatically by `Foundation` under the hood transparently for us?) or using a `FileInputStream` to read data as needed instead of loading it all could help on that matter. Though we'd have to experiment, because it might be that some operations we do on our data to be uploaded will end up loading the whole data in memory anyway even with `mmap` or `FileInputStream` (e.g. computing the checksum).